File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>


int main(int argc, string argv[])
{
    // Checks that th euser enterd 1 argument only (the key)
    if (argc != 2) {
        printf("Usage ./caesar key\n");
        return 1;
    } else {
        // checks that the key is digits only
        for (int i = 0, j = strlen(argv[1]); i < j; i++) {
            if (isdigit(argv[1][i]) == 0) {
                printf("Usage ./caesar key\n");
                return 1;
            }
        }
        int key = atoi(argv[1]);
        // Gets the plain text from the user
        string plainText = get_string("plaintext: ");
        // Encoding the plain text
        for (int i = 0, j = strlen(plainText); i < j; i++){
            if (isalpha(plainText[i]) != 0) {
                if (isupper(plainText[i]) != 0) {
                    plainText[i] = ((plainText[i] - 'A') + key) % 26 + 'A';
                } else {
                    plainText[i] = ((plainText[i] - 'a') + key) % 26 + 'a';
                }
            }
        }
        printf("ciphertext: %s\n", plainText);
        return 0;
    }
}

File: pset6/caesar/caesar.py
import cs50
import sys

# checks that the user entered 1 argument only (the key)
if len(sys.argv) != 2:
    print("Usage ./caesar key")
    exit(1)
else:
    # checks that the key is digits only
    if not sys.argv[1].isdigit():
        print("Usage ./caesar key")
        exit(1)

key = int(sys.argv[1])

# Gets the plain text from the user and save it as a list
# Because string are immutable in python and I want to change each letter with its encoded one
plainText = list(cs50.get_string("plaintext: "))

# Encoding the plain text
for i in range(len(plainText)):
    if plainText[i].isalpha():
        if  plainText[i].isupper():
            cipheredLetter = ((ord(plainText[i]) - ord('A')) + key) % 26 + ord('A')
            plainText[i] = chr(cipheredLetter)
        else:
            cipheredLetter = ((ord(plainText[i]) - ord('a')) + key) % 26 + ord('a')
            plainText[i] = chr(cipheredLetter)

# print the encrypted text
print("ciphertext: " + "".join(plainText))